### PR TITLE
Django - Add Support for Storing Users Not in Guild

### PR DESCRIPTION
**Pull request by MarkKoz**
As discussed in  #149, we need the `Members` (now renamed to `Users`) to facilitate users not in our guild.

I think a simple boolean field is all that is needed. The only field that is unique to members is `roles`, but that may be valuable to keep track of in the event a member leaves the guild and rejoins. I made `roles` not required, but I am not sure if that is the best thing to do.

### CODE REVIEW

File: api/admin.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from django.contrib import admin
 
 from .models import (
-    DocumentationLink, Member,
+    DocumentationLink, User,
```
Comment (jchristgit) [inline api/admin.py]: Doesn't cornflakes complain about this?

File: api/admin.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 
 admin.site.register(DocumentationLink)
-admin.site.register(Member)
+admin.site.register(User)
```
Comment (jchristgit) [inline api/admin.py]: Would be nice to keep this in order.

File: api/serializers.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 from .models import (
     DocumentationLink,
-    Member, OffTopicChannelName,
+    User, OffTopicChannelName,
```
Comment (jchristgit) [inline api/serializers.py]: the cornflakes they will shutter

File: api/tests/test_models.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from django.test import SimpleTestCase
 
 from ..models import (
-    DocumentationLink, Member, ModelReprMixin,
+    DocumentationLink, User, ModelReprMixin,
```
Comment (jchristgit) [inline api/tests/test_models.py]: the f̤͍͍́l͖̣͙̥ạke͇̳̘̤s͏͍͈̝̳ must n̷o̕t be a̧͞ǹ̴g̨̀͝ere͘d̵͟

File: api/tests/test_users.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 from django_hosts.resolvers import reverse
 
 from .base import APISubdomainTestCase
-from ..models import Member, Role
+from ..models import User, Role
```
Comment (jchristgit) [inline api/tests/test_users.py]: a͢͏͟a̛͞ąa̕aa help

File: api/urls.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 from .views import HealthcheckView
 from .viewsets import (
-    DocumentationLinkViewSet, MemberViewSet,
+    DocumentationLinkViewSet, UserViewSet,
```
Comment (jchristgit) [inline api/urls.py]: the f̴͡͞l͘a̶ḱe͡͞͝man h̐ͤ̍̃ͯͩͭê̵̑̿ͣͨ ̡̌̉c̀́oͭ́ͫ̒ͭ̓̃͞m̡̅̑ͦ̄͒͗e̢ͣ͆̒ͭ̅́s̛̅

File: api/viewsets.py [hunk 7]
```diff
@@ -10,15 +10,15 @@
 from rest_framework_bulk import BulkCreateModelMixin
 
 from .models import (
-    DocumentationLink, Member,
+    DocumentationLink, User,
```
Comment (jchristgit) [inline api/viewsets.py]: aaaaaaaa

File: api/viewsets.py [hunk 8]
```diff
@@ -10,15 +10,15 @@
...
     OffTopicChannelName,
     SnakeFact, SnakeIdiom,
     SnakeName, SpecialSnake,
     Tag
 )
 from .serializers import (
     DocumentationLinkSerializer,
-    MemberSerializer, OffTopicChannelNameSerializer,
+    UserSerializer, OffTopicChannelNameSerializer,
```
Comment (jchristgit) [inline api/viewsets.py]: I think you know the drill

**Review by jchristgit — COMMENTED** [review]
Looks good so far. I think the flakes will be mad, though.

Do you think we should have `roles` be `NULL` if `not is_member`?

**Comment by MarkKoz** [comment]
Right now, `roles` will just be an empty list. `null=True` has no effect on many to many relationships.

**Comment by jchristgit** [comment]
Ah, good point, I forgot about that.

**Review by jchristgit — APPROVED** [review]
Looks solid. Thanks for the PR!

**Pull request merged**